File: WebApplication1/WandSky.Core/Entitiies/Booking.cs
namespace WandSky.Core.Entities
{
    public class Booking : BaseEntity
    {
        public Guid UserId { get; set; }
        public string PackageId { get; set; } = string.Empty;
        public Guid PaymentId { get; set; }
        public BookingStatus Status { get; set; }
        public decimal TotalAmount { get; set; }
        public string Currency { get; set; } = "nzd";
        public string CustomerName { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string Phone { get; set; } = string.Empty;
        public DateTime TravelDate { get; set; }
        public int Travelers { get; set; }
        public string? SpecialRequests { get; set; }
        public DateTime? ConfirmedAt { get; set; }
        public DateTime? CancelledAt { get; set; }
        public string? CancellationReason { get; set; }
        public string BookingReference { get; set; } = string.Empty;

        // Navigation properties
        public User User { get; set; } = null!;
        public Payment Payment { get; set; } = null!;
    }

    public enum BookingStatus
    {
        Pending = 0,
        Confirmed = 1,
        InProgress = 2,
        Completed = 3,
        Cancelled = 4,
        Refunded = 5
    }
}

File: nz_tourism/types/booking.ts
// nz_tourism/types/booking.ts
export interface BookingFormData {
  customerName: string;
  email: string;
  phone: string;
  travelers: number;
  travelDate: string;
  specialRequests: string;
}

export interface Booking {
  id: string;
  bookingReference: string;
  packageId: string;
  packageTitle?: string;
  packageImage?: string;
  packageDuration?: string;
  customerName: string;
  email: string;
  phone?: string;
  travelers: number;
  travelDate: string;
  totalAmount: number;
  currency: string;
  status: BookingStatus;
  specialRequests?: string;
  createdAt: string;
  updatedAt?: string;
  confirmedAt?: string;
  cancelledAt?: string;
  cancellationReason?: string;
}

export type BookingStatus = 'Confirmed' | 'Cancelled' | 'Pending' | 'Completed';

export interface CreateBookingRequest {
  paymentIntentId: string;
  packageId: string;
  bookingData: BookingFormData;
  totalAmount: number;
}

export interface BookingResponse {
  success: boolean;
  data?: Booking;
  error?: string;
}

export interface BookingListResponse {
  success: boolean;
  data?: Booking[];
  error?: string;
  total?: number;
  page?: number;
  pageSize?: number;
}

// 预订状态配置
export const BOOKING_STATUS_CONFIG = {
  'Confirmed': {
    label: '已确认',
    color: 'confirmed',
    bgColor: '#d1fae5',
    textColor: '#065f46'
  },
  'Pending': {
    label: '待确认',
    color: 'pending',
    bgColor: '#fef3c7',
    textColor: '#92400e'
  },
  'Cancelled': {
    label: '已取消',
    color: 'cancelled',
    bgColor: '#fee2e2',
    textColor: '#991b1b'
  },
  'Completed': {
    label: '已完成',
    color: 'completed',
    bgColor: '#e0e7ff',
    textColor: '#3730a3'
  }
} as const;